### PR TITLE
Fixes #3700 - cleanup external volume validation

### LINKED ISSUE #3700: External Volumes: Clean up validation
The current state of the validation logic is a bit overcomplicated. We also need to verify that the resulting error paths make sense.

Depends on #3698.

Related epic (sorry, internal): [MARATHON-749](https://mesosphere.atlassian.net/browse/MARATHON-749)

### PR DESCRIPTION
**Pull request by kolloch**

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/core/externalvolume/impl/providers/DVDITest.scala [hunk 1]
```diff
@@ -68,6 +69,8 @@ class DVDIProviderVolumeValidationTest extends MarathonSpec with Matchers with T
   for ((tc, idx) <- ttValidateVolume.zipWithIndex; (v, vidx) <- tc.volumes.zipWithIndex) {
     test(s"validExternalVolume $idx,$vidx") {
       val result = validate(v)(DVDIProvider.validations.volume)
+      val constrains = ValidationHelper.getAllRuleConstrains(result)
```
Comment (jdef) [inline src/test/scala/mesosphere/marathon/core/externalvolume/impl/providers/DVDITest.scala]: where is `constrains` used?

Comment (kolloch) [inline src/test/scala/mesosphere/marathon/core/externalvolume/impl/providers/DVDITest.scala]: left-over from a debugging session, thanks

Comment (philipnrmn) [inline src/test/scala/mesosphere/marathon/core/externalvolume/impl/providers/DVDITest.scala]: It should `constraints` in any case :)

**Comment by aquamatthias** [comment]
Thanks!

**Pull request merged**